sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py

```diff
@@ -147,14 +146,10 @@ def __init__(
         self._query_str, credential = self._format_query_string(sas_token, credential)
         super(ContainerClient, self).__init__(parsed_url, service='blob', credential=credential, **kwargs)
         self._api_version = get_api_version(kwargs)
-        self._client = self._build_generated_client()
+        self._client = AzureBlobStorage(self.url, base_url=self.url, pipeline=self._pipeline)
+        self._client._config.version = self._api_version # pylint: disable=protected-access
         self._configure_encryption(kwargs)
 
-    def _build_generated_client(self):
```

Comment:
Could you bring this back? I believe I added this when we needed to build a generated client twice. I think it's better to have only one place where we build the generated client, so we don't accidentally miss one instance when we change something. It's fine to have it twice, once in sync and once in async (and not in helper file). 

You should be able to type it `def _build_generated_client(self) -> AzureBlobStorage`: